static/app/views/projectInstall/messagingIntegrationAlertRule.tsx

```diff
@@ -0,0 +1,120 @@
+import styled from '@emotion/styled';
+
+import SelectControl from 'sentry/components/forms/controls/selectControl';
+import Input from 'sentry/components/input';
+import {t} from 'sentry/locale';
+import {space} from 'sentry/styles/space';
+import type {OrganizationIntegration} from 'sentry/types/integrations';
+import {
+  type IssueAlertNotificationProps,
+  providerDetails,
+} from 'sentry/views/projectInstall/issueAlertNotificationOptions';
+
+type Props = {
+  notificationProps: IssueAlertNotificationProps;
+  providersToIntegrations: Record<string, OrganizationIntegration[]>;
+};
+
+export default function MessagingIntegrationAlertRule({
+  notificationProps,
+  providersToIntegrations,
+}: Props) {
+  const {channel, integration, provider, setChannel, setIntegration, setProvider} =
+    notificationProps;
+
+  const providerOptions = useMemo(
+    () =>
+      Object.keys(providersToIntegrations).map(p => ({
+        value: p,
+        label: providerDetails[p].name,
+      })),
+    [providersToIntegrations]
+  );
+  const integrationOptions = useMemo(
+    () =>
+      provider && providersToIntegrations[provider]
+        ? providersToIntegrations[provider]?.map(i => ({
+            value: i,
+            label: i.name,
+          }))
+        : [],
+    [providersToIntegrations, provider]
+  );
+
+  useEffect(() => {
+    const providerKeys = Object.keys(providersToIntegrations);
+    if (providerKeys.length > 0) {
+      const firstProvider = providerKeys[0];
+      setProvider(firstProvider);
+
+      const firstIntegration = providersToIntegrations[firstProvider][0];
+      setIntegration(firstIntegration);
+    }
+  }, [providersToIntegrations, setProvider, setIntegration]);
+
+  if (!provider) {
+    return null;
+  }
+
+  return (
+    <div>
+      <RuleWrapper>
+        <Rule>
+          {t('Send')}
+          <InlineSelectControl
+            label="provider"
+            disabled={Object.keys(providersToIntegrations).length === 1}
+            value={provider}
+            options={providerOptions}
+            onChange={p => {
+              setProvider(p.value);
+              setIntegration(providersToIntegrations[p.value][0]);
+              setChannel('');
+            }}
+          />
+          {t('notification to the')}
+          <InlineSelectControl
+            label="integration"
+            role="select"
+            disabled={integrationOptions.length === 1}
+            value={integration}
+            options={integrationOptions}
+            onChange={i => setIntegration(i.value)}
+          />
+          {providerDetails[provider]?.label}
+          <InlineInput
+            // label="channel"
+            type="text"
+            value={channel || ''}
+            placeholder={providerDetails[provider]?.placeholder}
+            onChange={(e: React.ChangeEvent<HTMLInputElement>) =>
+              setChannel(e.target.value)
+            }
+          />
+        </Rule>
```

Comment:
This whole thing actually needs to be using `tct` (translate string with component interpolation).

The reason for this is because the english sentence structure is not the same as other language sentence structures.

Right now this might produce something that looks like

> Send [Slack] notification to the [my slack] workspace to [#alerts]

But now if we take that sentence and look at what it looks like in Japanese you get

> [Slack]通知を[my slack]ワークスペースに[#alerts]に送信します

Right away you can notice that the ordering is a little bit different.

In Japanese the sentence uses Subject Object Verb ordering, unlike english that uses subject verb object.

In Japanese "[Slack]通知" is "Slack Notification", then at the end we have "送信します" which roughly means "send / transmit".

However the way this is written right now, the sentence structure of this sentence and these components cannot be moved around like that. In fact, it's even worse because the translator is only given two strings to trnaslate

`Send` and `notification to the`. Out of context in other languages it's impossible to know what additional gramatical markers should be includd.

Now, if you were to change this to use `tct` we can represent the whole sentence.

It's a little tricky for this example, because `providerDetails[provider]?.label` is actually also part of the sentence and should be translated. So what we probably want is that instead of having the provider configuration look like this:

```tsx
  slack: {
    name: 'Slack',
    action: IssueAlertActionType.SLACK,
    label: 'workspace to',
    placeholder: 'channel, e.g. #critical',
  },
  discord: {
    name: 'Discord',
    action: IssueAlertActionType.DISCORD,
    label: 'server in the channel',
    placeholder: 'channel ID or URL',
  },
```

It should look closer to

```tsx
  slack: {
    name: t('Slack'),
    action: IssueAlertActionType.SLACK,
    placeholder: t('channel, e.g. #critical'),
    makeSentence: ({providerName, integrationName, target}) => tct(
      'Send [providerName] notification to the [integrationName] workspace to [target]',
      {
        providerName,
        integrationName,
        target
      }
    )
  },
  discord: {
    name: t('Discord'),
    action: IssueAlertActionType.DISCORD,
    placeholder: t('channel ID or URL'),
    makeSentence: ({providerName, integrationName, target}) => tct(
      'Send [providerName] notification to the [integrationName] server in the channel [target]',
      {
        providerName,
        integrationName,
        target
      }
    )
  },
```

now the translator will have the string

> Send [providerName] notification to the [integrationName] workspace to [target]

If the translator translates this to Japanese it will look like this:

> [providerName]通知を[integrationName]ワークスペース、[target]に送信します

See how they were able to move around the object and the verb. That wasn't possible before.

Now this code looks like

```suggestion
        <Rule>
          {providerDetails[provider]?.makeSentence({
            providerName: (
	            <InlineSelectControl
		            label="provider"
		            disabled={Object.keys(providersToIntegrations).length === 1}
		            value={provider}
		            options={providerOptions}
		            onChange={p => {
		              setProvider(p.value);
		              setIntegration(providersToIntegrations[p.value][0]);
		              setChannel('');
		            }}
		          />
		        ),
		        integrationName: (
		          <InlineSelectControl
		            label="integration"
		            role="select"
		            disabled={integrationOptions.length === 1}
		            value={integration}
		            options={integrationOptions}
		            onChange={i => setIntegration(i.value)}
		          />
		        ),
		        target: (
		          <InlineInput
		            // label="channel"
		            type="text"
		            value={channel || ''}
		            placeholder={providerDetails[provider]?.placeholder}
		            onChange={(e: React.ChangeEvent<HTMLInputElement>) =>
		              setChannel(e.target.value)
		            }
		          />
		        ),
		      }) ?? tct('Select via [providerName]', {providerName: (...)})}
        </Rule>
```

This can now be correctly translated across languages